pmd-java/etc/grammar/Java.jjt

```diff
@@ -1751,9 +1803,9 @@ void EnumBody():
 }
 
 void EnumConstant():
-{Token t;}
+{}
 {
-  t=<IDENTIFIER> {jjtThis.setImage(t.image);} [ Arguments() ] [ ClassOrInterfaceBody() ]
+  VariableDeclaratorId() [ ArgumentList() ] [ ClassOrInterfaceBody() #AnonymousClassDeclaration ]
```

Comment:
`VariableDeclaratorId()` checks that the name is not a keyword like `enum` which is nice.  The production is needlessly complicated because it's used also for receiver parameters, which I now think was not a good idea. We could rework that in the future
 
Pushing a VariableDeclaratorId allows all things that declare a value to uniformly be represented by that node. This makes things easier for a symbol table, or for rules. Enum constants are after all just sugared field declarations.

Reply:
The annotations are just laid out between constants with no structure, it would be nice to correct that later by nesting them within the node

Reply:
Ah, I've seen now where the annotations are handled (in EnumBody) - that was my concern :)
I'll add the point with the explicit receiver parameters to the TODO list, though.